bitcoin/src/blockdata/fee_rate.rs

```diff
@@ -136,3 +136,67 @@ impl Div<Weight> for Amount {
 }
 
 crate::parse::impl_parse_str_from_int_infallible!(FeeRate, u64, from_sat_per_kwu);
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn fee_rate_const_test() {
+        assert_eq!(0, FeeRate::ZERO.to_sat_per_kwu());
+        assert_eq!(u64::min_value(), FeeRate::MIN.to_sat_per_kwu());
+        assert_eq!(u64::max_value(), FeeRate::MAX.to_sat_per_kwu());
+        assert_eq!(250, FeeRate::BROADCAST_MIN.to_sat_per_kwu());
+        assert_eq!(750, FeeRate::DUST.to_sat_per_kwu());
+    }
+
+    #[test]
+    fn fee_rate_from_sat_per_vb_test() {
+        let fee_rate = FeeRate::from_sat_per_vb(10).expect("expected feerate in sat/kwu");
+        assert_eq!(FeeRate(2500), fee_rate);
+    }
+
+    #[test]
+    fn fee_rate_from_sat_per_vb_overflow_test() {
+        let fee_rate = FeeRate::from_sat_per_vb(std::u64::MAX);
+        assert!(fee_rate.is_none());
+    }
+
+    #[test]
+    fn from_sat_per_vb_unchecked_test() {
+        let fee_rate = FeeRate::from_sat_per_vb_unchecked(10);
+        assert_eq!(FeeRate(2500), fee_rate);
+    }
+
+    #[test]
+    #[should_panic]
+    fn from_sat_per_vb_unchecked_panic_test() {
+        FeeRate::from_sat_per_vb_unchecked(std::u64::MAX);
```

Comment:
What is the motivation for using the fully qualified path to `u64`?

Reply:
Good point.  `std` can be removed since it's already in scope.  thanks!

Reply:
Now I remember why I had included the fqp.  With Rust `1.41` I get the following message:

```
error[E0599]: no associated item named `MAX` found for type `u64` in the current scope
   --> bitcoin/src/blockdata/fee_rate.rs:174:49
    |
174 |         FeeRate::from_sat_per_vb_unchecked(u64::MAX);
    |                                                 ^^^ associated item not found in `u64`
```

Reply:
Added `use std::u64;` to the tests for the 1.41 toolchain.

Reply:
Oh my bad, I should have noticed that. In Rust 1.41.1 we need to use `max_value()` instead of the constant `MAX`.

Reply:
https://doc.rust-lang.org/std/primitive.u64.html#associatedconstant.MAX

`MAX` only came in in Rust 1.43

Reply:
I'm confused now why this passes CI with the import you mention.

Reply:
It passes because it using the constants for u64 https://doc.rust-lang.org/std/u64/index.html.  The constants are deprecated but as you say, `MAX` isn't available until Rust `1.43`.

Reply:
If CI passes I'm happy to merge it, thanks.